flink-java/src/main/java/org/apache/flink/api/java/DataSet.java

```diff
@@ -181,7 +181,17 @@ protected void fillInType(TypeInformation<T> typeInfo) {
 		return this.type;
 	}
 
+	/**
+     *  1. Check if the function is implemented by a scala object. Checks only if scala object function forbidden
+     *    is not disabled in the [[org.apache.flink.api.common.ExecutionConfig]]
+     *
+     *  2. Returns a "closure-cleaned" version of the given function. Cleans only if closure cleaning
+     *     is not disabled in the [[org.apache.flink.api.common.ExecutionConfig]]
+     */
 	public <F> F clean(F f) {
```

Comment:
I suggest to move this check in a separate method like:

```public <F> F checkNotSingleton(F f)```

One method should typically have one single duty, and the duty of a cleaning function should not be checking for singleton objects. Both functions could then be chained however into `clean(check(f))`.